FlowCrypt/Functionality/DataManager/Encrypted Storage/SchemaMigration.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  SchemaMigration.swift
+//  FlowCrypt
+//
+//  Created by  Ivan Ushakov on 08.12.2021
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import RealmSwift
+
+protocol RealmProperty {
+    var propertyKey: String { get }
+}
+
+enum SchemaMigration {}
+
+extension SchemaMigration {
+    enum Properties {}
+}
+
+extension Object {
+    subscript(property: RealmProperty) -> Any? {
+        get {
+            return self[property.propertyKey]
+        }
+        set {
+            self[property.propertyKey] = newValue
+        }
+    }
+}
+
+extension SchemaMigration.Properties {
+    enum User: String, RealmProperty {
```

Comment:
I did this to avoid string duplication. Now we keep all Realm property names statically typed.

Reply:
I don't really understand this - I think @sosnovsky may better be able to have a look.

Reply:
It's a good idea to have enums for all database properties - it'll help to avoid typos when typing object properties in migrations. Also property names will have autocompletion, which is useful too.

I see a couple possible improvements here:
- what about using `rawValue` instead of custom `propertyKey` property? So we won't need `var propertyKey: String { ...` in every `RealmProperty` and protocol will look like this
```swift
protocol RealmProperty {
     var rawValue: String { get }
}
```

- also this code looks a bit messy
```swift
enum SchemaMigration {}

 extension SchemaMigration {
     enum Properties {}
 }

 extension SchemaMigration.Properties {
...
```
Can we use just simple struct like this?
```swift
struct SchemaMigration {
    struct Properties {
        enum User: String, RealmProperty {
...
```